paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -335,73 +367,92 @@ def sync_boto_secrets(
                     log.warning(
                         f"Boto key {this_key} required for {service} could not be found."
                     )
+
         if not secret_data:
             continue
+
         # In order to prevent slamming the k8s API, add some artificial delay here
         time.sleep(0.3)
-        app_name = get_kubernetes_app_name(service, instance)
-        namespace = instance_config.get_namespace()
-        secret = limit_size_with_hash(f"paasta-boto-key-{app_name}")
-        hashable_data = "".join([secret_data[key] for key in secret_data])
-        signature = hashlib.sha1(hashable_data.encode("utf-8")).hexdigest()
-        kubernetes_signature = get_kubernetes_secret_signature(
-            kube_client=kube_client,
-            secret=secret,
+        update_k8s_secret(
             service=service,
-            namespace=namespace,
+            secret_name=instance_config.get_boto_k8s_secret_name(),
+            secret_data=secret_data,
+            secret_signature=get_dict_signature(secret_data),
+            kube_client=kube_client,
+            namespace=instance_config.get_namespace(),
         )
-        if not kubernetes_signature:
-            log.info(f"{secret} for {service} in {namespace} not found, creating")
-            try:
-                create_plaintext_dict_secret(
-                    kube_client=kube_client,
-                    secret_name=secret,
-                    secret_data=secret_data,
-                    service=service,
-                    namespace=namespace,
-                )
-            except ApiException as e:
-                if e.status == 409:
-                    log.warning(
-                        f"Secret {secret} for {service} already exists in {namespace} but no signature found. Updating secret and signature."
-                    )
-                    update_plaintext_dict_secret(
-                        kube_client=kube_client,
-                        secret_name=secret,
-                        secret_data=secret_data,
-                        service=service,
-                        namespace=namespace,
-                    )
-                else:
-                    raise
-            create_kubernetes_secret_signature(
+    return True
+
+
+def get_dict_signature(data: Dict[str, str]) -> str:
+    return hashlib.sha1("".join(data.values()).encode("utf-8")).hexdigest()
```

Comment:
hmm, thinking about this a little more: should we also include the keys in the signature generation?

e.g.,
```
{
    "super-secret-encryption-key": "jahmdflkjahsdf", 
    "super-secret-decryption-key": "abc",
}
```
is not the same secret data as 
```
    "super-secret-decryption-key": "jahmdflkjahsdf", 
    "super-secret-encryption-key": "abc",
```

also, might be nice to either add a docstring for this function or mark it as private (e.g., `_get_dict_signature`) since this isn't really a function we want folks to be importing as-is

Reply:
How should keys and values be joined? Should we include some kind of separator otherwise there's still chance of overlapping signatures

Reply:
maybe something like this?
```Python
def _get_dict_signature(data: Dict[str, str]) -> str:
    unhashed_contents = (f"{key}:{value}" for key, value in data.items())
    return hashlib.sha1("|".join(unhashed_contents).encode("utf-8")).hexdigest()
```